# SETUP

In [78]:
SELECT * FROM dbo.covid_deaths_trincan
WHERE (location='Canada' OR location='Trinidad and Tobago') 
AND total_deaths IS NOT NULL 
AND icu_patients IS NOT NULL 
AND hosp_patients IS NOT NULL 
AND icu_patients_per_million IS NOT NULL


(1358 rows affected)

Total execution time: 00:00:00.171

iso_code continent location date population total_cases new_cases new_cases_smoothed total_deaths new_deaths new_deaths_smoothed total_cases_per_million new_cases_per_million new_cases_smoothed_per_million total_deaths_per_million new_deaths_per_million new_deaths_smoothed_per_million icu_patients icu_patients_per_million hosp_patients CAN North America Canada 01,04,2020 38454328 7424 1169 825.429 56 14 5.429 193.06 30.4 21.465 1.456 0.364 0.141 203 5.279 836 CAN North America Canada 02,04,2020 38454328 8536 1112 939.571 63 7 6.286 221.978 28.917 24.433 1.638 0.182 0.163 222 5.773 571 CAN North America Canada 03,04,2020 38454328 9595 1059 887.143 72 9 7.143 249.517 27.539 23.07 1.872 0.234 0.186 225 5.851 593 CAN North America Canada 04,04,2020 38454328 11268 1673 1035.714 138 66 16.286 293.023 43.506 26.934 3.589 1.716 0.424 230 5.981 664 CAN North America Canada 05,04,2020 38454328 12519 1251 1120.571 187 49 21.714 325.555 32.532 29.14 4.863 1.274 0.565 250 6.501 687 CAN North America Canada 06,04,2020 38454328 13882 1363 1213.714 231 44 27.143 361 35.445 31.562 6.007 1.144 0.706 433 11.26 1271 CAN North America Canada 07,04,2020 38454328 15496 1614 1320.143 280 49 34 402.972 41.972 34.33 7.281 1.274 0.884 461 11.988 1354 CAN North America Canada 08,04,2020 38454328 16653 1157 1318.429 323 43 38.143 433.059 30.088 34.286 8.4 1.118 0.992 493 12.82 1444 CAN North America Canada 09,04,2020 38454328 17883 1230 1335.286 379 56 45.143 465.045 31.986 34.724 9.856 1.456 1.174 544 14.147 1654 CAN North America Canada 10,04,2020 38454328 19274 1391 1382.714 434 55 51.714 501.218 36.173 35.957 11.286 1.43 1.345 498 12.95 1716 CAN North America Canada 11,04,2020 38454328 20748 1474 1354.286 508 74 52.857 539.549 38.331 35.218 13.21 1.924 1.375 509 13.236 1775 CAN North America Canada 12,04,2020 38454328 22133 1385 1373.429 568 60 54.429 575.566 36.017 35.716 14.771 1.56 1.415 519 13.497 1832 CAN North America Canada 13,04,2020 38454328 23301 1168 1345.571 652 84 60.143 605.94 30.374 34.991 16.955 2.184 1.564 500 13.002 1911 CAN North America Canada 14,04,2020 38454328 24365 1064 1267 716 64 62.286 633.609 27.669 32.948 18.619 1.664 1.62 529 13.757 1955 CAN North America Canada 15,04,2020 38454328 25663 1298 1287.143 779 63 65.143 667.363 33.754 33.472 20.258 1.638 1.694 522 13.575 2039 CAN North America Canada 16,04,2020 38454328 27046 1383 1309 902 123 74.714 703.328 35.965 34.04 23.456 3.199 1.943 527 13.705 2074 CAN North America Canada 17,04,2020 38454328 28364 1318 1298.571 1009 107 82.143 737.602 34.274 33.769 26.239 2.783 2.136 525 13.653 2139 CAN North America Canada 18,04,2020 38454328 30081 1717 1333.286 1191 182 97.571 782.253 44.65 34.672 30.972 4.733 2.537 515 13.393 2150 CAN North America Canada 19,04,2020 38454328 31872 1791 1391.286 1306 115 105.429 828.827 46.575 36.18 33.962 2.991 2.742 534 13.887 2271 CAN North America Canada 20,04,2020 38454328 33340 1468 1434.143 1464 158 116 867.003 38.175 37.295 38.071 4.109 3.017 530 13.783 2343 CAN North America Canada 21,04,2020 38454328 34776 1436 1487.286 1577 113 123 904.346 37.343 38.677 41.01 2.939 3.199 524 13.627 2406 CAN North America Canada 22,04,2020 38454328 36822 2046 1594.143 1687 110 129.714 957.552 53.206 41.455 43.87 2.861 3.373 516 13.419 2560 CAN North America Canada 23,04,2020 38454328 38412 1590 1623.714 1831 144 132.714 998.899 41.348 42.224 47.615 3.745 3.451 541 14.069 2632 CAN North America Canada 24,04,2020 38454328 40179 1767 1687.857 1971 140 137.429 1044.85 45.951 43.893 51.256 3.641 3.574 525 13.653 2689 CAN North America Canada 25,04,2020 38454328 42099 1920 1716.857 2143 172 136 1094.779 49.929 44.647 55.728 4.473 3.537 525 13.653 2708 CAN North America Canada 26,04,2020 38454328 43877 1778 1715 2299 156 141.857 1141.016 46.237 44.598 59.785 4.057 3.689 519 13.497 2745 CAN North America Canada 27,04,2020 38454328 45341 1464 1714.429 2461 162 142.429 1179.087 38.071 44.584 63.998 4.213 3.704 520 13.523 2855 CAN North America Canada 28,04,2020 38454

# Total Cases vs Total Deaths

### Likelihood of dying if you contract covid in your country

\---\> Canada had a marginally greater overall mortality rate than Trinidad.

In [79]:
WITH MortalityRanked AS (
    SELECT
        location, date, total_cases, new_cases, total_deaths,
        (CONVERT(float, total_deaths) / NULLIF(CONVERT(float, total_cases), 0)) * 100 AS mortality,
        RANK() OVER (PARTITION BY location ORDER BY (CONVERT(float, total_deaths) / NULLIF(CONVERT(float, total_cases), 0)) * 100 DESC) AS rank_order
    FROM
        dbo.covid_deaths_trincan
    WHERE
        (location='Canada' OR location LIKE '%Trinidad%')
)
SELECT
    location, date, total_cases, new_cases, total_deaths, mortality
FROM
    MortalityRanked
WHERE
    rank_order <= 5
ORDER BY
    location, rank_order;


(11 rows affected)

Total execution time: 00:00:00.497

location date total_cases new_cases total_deaths mortality Canada 22,06,2020 101008 390 8381 8.297362585141771 Canada 23,06,2020 101326 318 8398 8.288099796695814 Canada 24,06,2020 101626 300 8404 8.269537323125972 Canada 21,06,2020 100618 409 8318 8.266910493152318 Canada 26,06,2020 102231 279 8451 8.266572761686767 Trinidad and Tobago 07,04,2020 105 5 8 7.6190476190476195 Trinidad and Tobago 08,04,2020 106 1 8 7.547169811320755 Trinidad and Tobago 09,04,2020 107 1 8 7.476635514018691 Trinidad and Tobago 10,04,2020 109 2 8 7.339449541284404 Trinidad and Tobago 11,04,2020 109 0 8 7.339449541284404 Trinidad and Tobago 12,04,2020 109 0 8 7.339449541284404

# Highest Infection Rate compared to population

### Trinidad:

**Rank #128**

Highest infection 191496 with infection percentage 12.507552041320851

### Canada:

**Rank #130**

Highest infection 38454328 with infection percentage 12.336689383832166

In [80]:
SELECT location, population, MAX(total_cases) AS highest_infection, MAX(CONVERT(float, total_cases) / CONVERT(float, population))*100 as infection_percent
FROM dbo.covid_deaths_trincan
GROUP BY location, population
ORDER BY infection_percent desc

(244 rows affected)

Total execution time: 00:00:00.204

location population highest_infection infection_percent San Marino 33690 25292 75.07272187592757 Cyprus 896007 660854 73.7554505712567 Brunei 449002 321235 71.54422474732853 Austria 8939617 6081287 68.02625884308019 South Korea 51815808 34571873 66.72070615978815 Faeroe Islands 53117 34658 65.24841387879586 Slovenia 2119843 1350022 63.68499931362841 Gibraltar 32677 20550 62.8882700370291 Martinique 367512 230354 62.67931387274429 Andorra 79843 48015 60.1367684080007 Jersey 110796 66391 59.92183833351385 Luxembourg 647601 387467 59.83113058812448 Saint Pierre and Miquelon 5885 3426 58.21580288870008 Denmark 5882259 3423176 58.19492137289433 France 67813000 38997490 57.507395337177236 Iceland 372903 209503 56.181634366041564 Guernsey 63329 35326 55.78171138025233 Portugal 10270857 5634044 54.8546630529468 Liechtenstein 39355 21561 54.78592300851226 Greece 10384972 5494594 52.90908824790284 Latvia 1850654 977068 52.79582244979343 Guadeloupe 395762 203235 51.352833268479536 Israel 9449000 4841772 51.24110487882315 Reunion 974062 494595 50.7765419449686 Switzerland 8740471 4436359 50.756521015858304 Falkland Islands 3801 1923 50.59194948697711 Singapore 5637022 2827115 50.15263378429249 Saint Barthelemy 10994 5507 50.09095870474805 Georgia 3744385 1855289 49.54856404990405 Lithuania 2750058 1359230 49.42550302575437 Netherlands 17564020 8623210 49.09587896165001 Seychelles 107135 51220 47.80883931488309 Bahrain 1472237 696614 47.31670240593056 Monaco 36491 17181 47.082842344687734 New Zealand 5185289 2434052 46.94149159285047 Germany 83369840 38437756 46.10510947364179 Niue 1952 895 45.85040983606557 Cayman Islands 68722 31472 45.79610605046419 Estonia 1326064 604950 45.61997007685903 Isle of Man 84534 38008 44.96179052215676 Czechia 10493990 4713739 44.91846285349996 Italy 59037472 26433512 44.77412582977808 Australia 26177410 11698780 44.69036470758566 Bonaire Sint Eustatius and Saba 27052 11922 44.070678692887775 Nauru 12691 5393 42.49468127019148 Cook Islands 17032 7218 42.37905119774542 Aruba 106459 44224 41.54087489080303 Belgium 11655923 4836886 41.49723706994289 European Union 450146793 184862903 41.06724870080325 Saint Helena 5401 2166 40.10368450286984 Montenegro 627082 251280 40.07131443734631 Barbados 281646 110141 39.10618293886652 Marshall Islands 41593 16138 38.79979804293992 Saint Martin (French part) 31816 12324 38.73522755846115 Puerto Rico 3252412 1252713 38.51643026775206 Serbia 6871547 2583470 37.596628532119475 United Kingdom 67508936 24812582 36.754514987467736 Maldives 523798 186694 35.64236595023272 Palau 18084 6249 34.55540809555408 Ireland 5023108 1726402 34.36919930847594 High income 1250514600 427084445 34.15269561826787 Europe 744807803 250942654 33.692269735793836 Slovakia 5643455 1871209 33.15715284342659 Oceania 45038860 14628086 32.47881052051495 French Guiana 304568 98041 32.19018412965249 Montserrat 4413 1403 31.792431452526625 Croatia 4030361 1279927 31.757130440672682 Wallis and Futuna 11596 3550 30.614004829251467 United States 338289856 103436829 30.57639097519968 Uruguay 3422796 1041111 30.416974894209297 Guam 171783 52187 30.379606829546578 Mongolia 3398373 1011388 29.760947370991943 Spain 47558632 13980340 29.396009540392164 Bermuda 64207 18860 29.37374429579329 Northern Mariana Islands 49574 14334 28.91435026425142 New Caledonia 289959 80064 27.612179652985425 Norway 5434324 1494968 27.509732581274136 Japan 123951696 33803572 27.27156875691318 Chile 19603736 5325080 27.16359779584871 Finland 5540745 1499712 27.06697384557492 Tuvalu 11335 2943 25.96382884869872 Sweden 10549349 2738097 25.955127657640297 French Polynesia 306292 78700 25.694435375393414 United States Virgin Islands 99479 25389 25.521969460891242 Sint Maarten (Dutch part) 44192 11051 25.00678855901521 Anguilla 15877 3904 24.589028153933363 Mauritius 1299478 315100 24.24819812263078 Curacao 191173 45883 24.00077416790028 Costa Rica 5180836 1230653 23.753946274307854 Panama 4408582 1041924 23.633993878303727 British Virgin Isla

# Total Population vs Vaccinations

### Canada's vaccine distribution was significantly faster than Trinidad

In [81]:
SELECT dea.continent, dea.location, CONVERT(DATE, REPLACE(dea.date, ',', '-'), 105) AS date, dea.population, vac.new_vaccinations,
SUM(CONVERT(int, vac.new_vaccinations)) OVER (PARTITION BY dea.location ORDER BY dea.location, dea.date) AS vac_count
FROM dbo.covid_deaths_trincan dea
JOIN dbo.covid_vaccines_trincan vac
    ON dea.location = vac.location
    AND dea.date = vac.date
WHERE dea.continent IS NOT NULL
AND (dea.location='Canada' OR dea.location LIKE '%Trinidad%')
ORDER BY 1, 2, 3


Warning: Null value is eliminated by an aggregate or other SET operation.

(2800 rows affected)

Total execution time: 00:00:00.760

continent location date population new_vaccinations vac_count North America Canada 2020-01-26 38454328 NULL 78927823 North America Canada 2020-01-27 38454328 NULL 81851420 North America Canada 2020-01-28 38454328 NULL 84865981 North America Canada 2020-01-29 38454328 NULL 88527864 North America Canada 2020-01-30 38454328 NULL 91834512 North America Canada 2020-01-31 38454328 NULL 94742322 North America Canada 2020-02-01 38454328 NULL 146854 North America Canada 2020-02-02 38454328 NULL 2822425 North America Canada 2020-02-03 38454328 NULL 5367034 North America Canada 2020-02-04 38454328 NULL 9299104 North America Canada 2020-02-05 38454328 NULL 12102788 North America Canada 2020-02-06 38454328 NULL 15169335 North America Canada 2020-02-07 38454328 NULL 18228870 North America Canada 2020-02-08 38454328 NULL 21488263 North America Canada 2020-02-09 38454328 NULL 24617494 North America Canada 2020-02-10 38454328 NULL 27835985 North America Canada 2020-02-11 38454328 NULL 31179375 North America Canada 2020-02-12 38454328 NULL 33847674 North America Canada 2020-02-13 38454328 NULL 37096215 North America Canada 2020-02-14 38454328 NULL 40326859 North America Canada 2020-02-15 38454328 NULL 43397916 North America Canada 2020-02-16 38454328 NULL 46670323 North America Canada 2020-02-17 38454328 NULL 50003498 North America Canada 2020-02-18 38454328 NULL 53357251 North America Canada 2020-02-19 38454328 NULL 56341181 North America Canada 2020-02-20 38454328 NULL 59594447 North America Canada 2020-02-21 38454328 NULL 62776743 North America Canada 2020-02-22 38454328 NULL 66293486 North America Canada 2020-02-23 38454328 NULL 69832411 North America Canada 2020-02-24 38454328 NULL 73251039 North America Canada 2020-02-25 38454328 NULL 76633330 North America Canada 2020-02-26 38454328 NULL 79238745 North America Canada 2020-02-27 38454328 NULL 82131260 North America Canada 2020-02-28 38454328 NULL 85140362 North America Canada 2020-02-29 38454328 NULL 88720545 North America Canada 2020-03-01 38454328 NULL 380063 North America Canada 2020-03-02 38454328 NULL 3027843 North America Canada 2020-03-03 38454328 NULL 5560430 North America Canada 2020-03-04 38454328 NULL 9478665 North America Canada 2020-03-05 38454328 NULL 12241034 North America Canada 2020-03-06 38454328 NULL 15288152 North America Canada 2020-03-07 38454328 NULL 18441648 North America Canada 2020-03-08 38454328 NULL 21628563 North America Canada 2020-03-09 38454328 NULL 24769583 North America Canada 2020-03-10 38454328 NULL 27999610 North America Canada 2020-03-11 38454328 NULL 31337617 North America Canada 2020-03-12 38454328 NULL 33968181 North America Canada 2020-03-13 38454328 NULL 37199715 North America Canada 2020-03-14 38454328 NULL 40409943 North America Canada 2020-03-15 38454328 NULL 43563149 North America Canada 2020-03-16 38454328 NULL 46783263 North America Canada 2020-03-17 38454328 NULL 50118797 North America Canada 2020-03-18 38454328 NULL 53468249 North America Canada 2020-03-19 38454328 NULL 56452386 North America Canada 2020-03-20 38454328 NULL 59699226 North America Canada 2020-03-21 38454328 NULL 62848007 North America Canada 2020-03-22 38454328 NULL 66491349 North America Canada 2020-03-23 38454328 NULL 69949126 North America Canada 2020-03-24 38454328 NULL 73359678 North America Canada 2020-03-25 38454328 NULL 76756719 North America Canada 2020-03-26 38454328 NULL 79351957 North America Canada 2020-03-27 38454328 NULL 82237413 North America Canada 2020-03-28 38454328 NULL 85526899 North America Canada 2020-03-29 38454328 NULL 88720545 North America Canada 2020-03-30 38454328 NULL 91992553 North America Canada 2020-03-31 38454328 NULL 95076005 North America Canada 2020-04-01 38454328 NULL 482790 North America Canada 2020-04-02 38454328 NULL 3137106 North America Canada 2020-04-03 38454328 NULL 5676473 North America Canada 2020-04-04 38454328 NULL 9614639 North America Canada 2020-04-05 38454328 NULL 12356547 North America Canada 2020-04-06 38454328 NULL 15